src/pmfind/pmfind.service.in

```diff
@@ -6,8 +6,8 @@ After=pmie_check.timer pmlogger_check.timer
 BindsTo=pmfind.timer
 
 [Service]
-Type=oneshot
-KillMode=none
+Type=simple
+Restart=no
```

Comment:
Restart=no is the default, so there isn't any need to specify this.

Type=simple means that the service will exit as soon as the main process has exited, i.e. the process started with ExecStart=.

I assume that you are trying to keep the original design where the main process exits but the children should remain running. This is not going to work like this, because systemd will kill those children when cleaning up, i.e. immediately after the main process has exited.

You have three options:
- change the design so that the main process waits for the children to exit before exiting itself
- change the design so that the main main process let's systemd think that the one of the children is the new main process, by setting Type=forking and either GuessMainPID=true or providing a file with the PID of one of the children. I don't know how many children there might be. This would require one of them to be guaranteed to run longer than the others.
- wait for https://github.com/systemd/systemd/issues/16805 to be implemented.

Please don't merge this as is.

(It doesn't matter here, since there are other issues, but `Type=simple` has been mostly obsoleted by `Type=exec`. Please use the latter whereever you'd use the former.)

Reply:
another option, use something like `systemd-run` to run the children, so they'd be their own separate services.

Reply:
Thanks for the comments - there is a bit of confusion here: not changing any of the PCP long running daemon services. They remain Type=notify. It's only the half hour pmlogger_check and nightly pmlogger_daily scripts (and similar for pmie and pmfind) services that are being changed. These services are started by a timer to perform regular house-keeping / log rolling etc and exit normally when completed, leaving no children. See new PR #1391 with improved commit messages. Not merging and closing this PR.